apps/backoffice-v2/src/pages/Entity/components/Case/hooks/usePendingRevisionEvents/usePendingRevisionEvents.tsx

```diff
@@ -0,0 +1,60 @@
+import { TWorkflowById } from '@/domains/workflows/fetchers';
+import { useCallback, useMemo } from 'react';
+import { calculateAllWorkflowEvents } from '@/pages/Entity/components/Case/hooks/usePendingRevisionEvents/utils/calculate-pending-workflow-events';
+import { CommonWorkflowEvent } from '@ballerine/common';
+import { checkIsKybExampleVariant } from '@/lib/blocks/variants/variant-checkers';
+import { useRevisionCaseMutation } from '@/domains/workflows/hooks/mutations/useRevisionCaseMutation/useRevisionCaseMutation';
+import { IPendingEvent } from './interfaces';
+
+function composeUniqueWorkflowEvents(
+  acc: Record<string, IPendingEvent>,
+  pendingWorkflowEvent: IPendingEvent,
+) {
+  acc[`${pendingWorkflowEvent?.workflowId}-${pendingWorkflowEvent?.pendingEvent}`] =
+    pendingWorkflowEvent;
+
+  return acc;
+}
+
+function isRevisionEvent(pendingWorkflowEvent: IPendingEvent) {
+  return (
+    pendingWorkflowEvent?.pendingEvent === CommonWorkflowEvent.REVISION ||
+    pendingWorkflowEvent?.pendingEvent === CommonWorkflowEvent.TASK_REVIEWED
+  );
+}
+
+export const usePendingRevisionEvents = (
+  mutateRevisionCase: ReturnType<typeof useRevisionCaseMutation>['mutate'],
+  workflow?: TWorkflowById,
+) => {
+  const pendingWorkflowEvents = useMemo(() => {
+    if (!workflow) return;
+
+    return calculateAllWorkflowEvents(workflow);
+  }, [workflow]);
+
+  const onMutateRevisionCase = useCallback(() => {
+    if (!pendingWorkflowEvents || !workflow) return;
+
+    const uniqueWorkflowEvents = pendingWorkflowEvents
+      .filter((pendingWorkflowEvent: IPendingEvent) => isRevisionEvent(pendingWorkflowEvent))
+      .reduce((acc: Record<string, IPendingEvent>, pendingWorkflowEvent: IPendingEvent) => {
+        return composeUniqueWorkflowEvents(acc, pendingWorkflowEvent);
+      }, {});
+
+    Object.keys(uniqueWorkflowEvents).forEach(pendingWorkflowKeys => {
+      const pendingWorkflowEvent = uniqueWorkflowEvents[pendingWorkflowKeys];
+      mutateRevisionCase({ workflowId: pendingWorkflowEvent!.workflowId });
+
+      const isKybExampleVariant = checkIsKybExampleVariant(workflow.workflowDefinition);
+      if (!isKybExampleVariant) return;
+
+      window.open(
+        `${workflow?.context?.metadata?.collectionFlowUrl}/?token=${pendingWorkflowEvent?.token}`,
+        pendingWorkflowEvent?.token,
```

Comment:
Can you explain this second argument?

Reply:
the second argument is "target"

when working with tabs.
without target it will always open a new tab.

with target. if the tab with the same target is already opened it will just switch you to the specific tab.